validator/src/admin_rpc_service.rs

```diff
@@ -1286,4 +1297,156 @@ mod tests {
+            .as_str()
+            .expect("Expected a string");
+        assert_eq!(
+            actual_validator_id,
+            expected_validator_id.pubkey().to_string()
+        );
+    }
+
+    // This test checks that `set_identity` call works with working validator and client.
+    #[test]
+    fn test_set_identity_with_validator() {
+        let leader_keypair = Keypair::new();
+        let leader_node = Node::new_localhost_with_pubkey(&leader_keypair.pubkey());
+
+        let validator_keypair = Keypair::new();
+        let validator_node = Node::new_localhost_with_pubkey(&validator_keypair.pubkey());
+        let genesis_config =
+            create_genesis_config_with_leader(10_000, &leader_keypair.pubkey(), 1000)
+                .genesis_config;
+        let (validator_ledger_path, _blockhash) = create_new_tmp_ledger!(&genesis_config);
+
+        let voting_keypair = Arc::new(Keypair::new());
+        let voting_pubkey = voting_keypair.pubkey();
+        let authorized_voter_keypairs = Arc::new(RwLock::new(vec![voting_keypair]));
+        let validator_config = ValidatorConfig {
+            rpc_addrs: Some((
+                validator_node.info.rpc().unwrap(),
+                validator_node.info.rpc_pubsub().unwrap(),
+            )),
+            ..ValidatorConfig::default_for_test()
+        };
+        let start_progress = Arc::new(RwLock::new(ValidatorStartProgress::default()));
+
+        let post_init = Arc::new(RwLock::new(None));
+        let meta = AdminRpcRequestMetadata {
+            rpc_addr: validator_config.rpc_addrs.map(|(rpc_addr, _)| rpc_addr),
+            start_time: SystemTime::now(),
+            start_progress: start_progress.clone(),
+            validator_exit: validator_config.validator_exit.clone(),
+            authorized_voter_keypairs: authorized_voter_keypairs.clone(),
+            tower_storage: Arc::new(NullTowerStorage {}),
+            post_init: post_init.clone(),
+            staked_nodes_overrides: Arc::new(RwLock::new(HashMap::new())),
+            rpc_to_plugin_manager_sender: None,
+        };
+
+        let _validator = Validator::new(
+            validator_node,
+            Arc::new(validator_keypair),
+            &validator_ledger_path,
+            &voting_pubkey,
+            authorized_voter_keypairs,
+            vec![leader_node.info],
+            &validator_config,
+            true, // should_check_duplicate_instance
+            None, // rpc_to_plugin_manager_receiver
+            start_progress.clone(),
+            SocketAddrSpace::Unspecified,
+            DEFAULT_TPU_USE_QUIC,
+            DEFAULT_TPU_CONNECTION_POOL_SIZE,
+            DEFAULT_TPU_ENABLE_UDP,
+            32, // max connections per IpAddr per minute for test
+            post_init,
+        )
+        .expect("assume successful validator start");
+        assert_eq!(
+            *start_progress.read().unwrap(),
+            ValidatorStartProgress::Running
+        );
```

Comment:
seems like we should have some test API that basically covers these setup activities and allows us to extract the RPC address

Reply:
Do you propose to extract a function for this setup?

Reply:
Got you. Yeah, i was also surprised. Maybe I was grepping wrong phrase but I could not find tests for this admin rpc.

Reply:
Updated